src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.Unix.cs

```diff
@@ -10,7 +11,7 @@ public partial class DebugProvider
     {
         private static readonly bool s_shouldWriteToStdErr = Environment.GetEnvironmentVariable("COMPlus_DebugWriteToStdErr") == "1";
 
-        public static void FailCore(string stackTrace, string message, string detailMessage, string errorSource)
+        public static void FailCore(string? stackTrace, string? message, string? detailMessage, string? errorSource)
```

Comment:
@maryamariyan, can you remind me why these DebugProvider.FailCore/WriteCore methods are public?

Reply:
DebugProvider and everything that comes with it is used by `System.Diagnostics.TraceSource` in CoreFX.